app/src/pages/inside/settingsPage/settingTabs/settingTabs.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { redirect } from 'redux-first-router';
+import { defineMessages, injectIntl, intlShape } from 'react-intl';
+import { PROJECT_SETTINGS_TAB_PAGE } from 'controllers/pages';
+import {
+  GENERAL,
+  NOTIFICATIONS,
+  BTS,
+  DEFECT,
+  ANALYSIS,
+  DEMO_DATA,
+} from 'common/constants/settingTabs';
+import { activeProjectSelector } from 'controllers/user';
+import { NavigationTabs } from 'components/main/navigationTabs';
+import classNames from 'classnames/bind';
+import { GeneralTab } from './generalTab';
+import styles from './settingTabs.scss';
+
+const cx = classNames.bind(styles);
+const messages = defineMessages({
+  general: {
+    id: 'SettingTabs.general',
+    defaultMessage: 'General',
+  },
+  notifications: {
+    id: 'SettingTabs.notifications',
+    defaultMessage: 'Notifications',
+  },
+  bts: {
+    id: 'SettingTabs.bts',
+    defaultMessage: 'Bug tracking system',
+  },
+  defect: {
+    id: 'SettingTabs.defect',
+    defaultMessage: 'Defect types',
+  },
+  analysis: {
+    id: 'SettingTabs.analysis',
+    defaultMessage: 'Auto-Analysis',
+  },
+  demoData: {
+    id: 'SettingTabs.demoData',
+    defaultMessage: 'Demo data',
+  },
+  beta: {
+    id: 'SettingTabs.beta',
+    defaultMessage: 'beta',
+  },
+});
+
+@connect(
+  (state) => ({
+    projectId: activeProjectSelector(state),
+    activeTab: state.location.payload.settingTab,
+  }),
+  {
+    changeTab: (link) => redirect(link),
+  },
+)
+@injectIntl
+export class SettingTabs extends Component {
+  static propTypes = {
+    projectId: PropTypes.string.isRequired,
+    intl: intlShape.isRequired,
+    changeTab: PropTypes.func.isRequired,
+    activeTab: PropTypes.string,
+  };
+  static defaultProps = {
+    activeTab: GENERAL,
+  };
+
+  tabsConfig = {
+    general: {
+      name: this.props.intl.formatMessage(messages.general),
+      link: {
```

Comment:
All of these links lead to the same page, the only difference is `settingTab` parameter. I've just got an idea: please make a function `createTabLink` which accepts destination tab as an argument (`tabName`) and use this function to create the config. This will make config more clear.